event-schema/event_schema_api.go

```diff
@@ -54,6 +56,181 @@ func (manager *EventSchemaManagerT) GetEventModels(w http.ResponseWriter, r *htt
 	w.Write(eventTypesJSON)
 }
 
+func (manager *EventSchemaManagerT) GetJsonSchemas(w http.ResponseWriter, r *http.Request) {
+	err := handleBasicAuth(r)
+	if err != nil {
+		http.Error(w, response.MakeResponse(err.Error()), 400)
+		return
+	}
+
+	if r.Method != http.MethodGet {
+		http.Error(w, response.MakeResponse("Only HTTP GET method is supported"), 400)
+		return
+	}
+
+	writeKeys, ok := r.URL.Query()["WriteKey"]
+	writeKey := ""
+	if ok && writeKeys[0] != "" {
+		writeKey = writeKeys[0]
+	}
+
+	eventTypes := manager.fetchEventModelsByWriteKey(writeKey)
+
+	// generating json schema from eventTypes
+	jsonSchemas, err := generateAllJsonSch(eventTypes)
+	if err != nil {
+		http.Error(w, response.MakeResponse("Internal Error: Failed to Marshal event types"), 500)
+		return
+	}
+
+	w.Write(jsonSchemas)
+}
+
+type PropertyTypeT struct {
+	Type []string `json:"type"`
+}
+
+type PropertiesT struct {
+	Required []string `json:"required"`
+	Property map[string]interface{} `json:"properties"`
+}
+
+func generateAllJsonSch(eventModels []*EventModelT)([]byte, error) {
+	jsonSchs := []map[string]interface{}{}
+	for _, eventModel := range eventModels {
+		flattenedSchema := make(map[string]interface{})
+		err := json.Unmarshal(eventModel.Schema, &flattenedSchema)
+		if err!=nil {
+		pkgLogger.Errorf("Unable to unmarshal eventModelSch")
+		}
+		unFlattened, err := unflatten(flattenedSchema)
+		if err!= nil {
+		}
+		filtered, ok := unFlattened["properties"].(map[string]interface{})
```

Comment:
are we generating schemas only for track events? what about identify etc.

Reply:
Done.